core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -41,7 +41,7 @@
 import org.apache.iceberg.types.Types;
 import org.apache.iceberg.util.ParallelIterable;
 
-class ManifestGroup {
+public class ManifestGroup {
```

Comment:
If we use scan API, we may not need to expose ManifestGroup as public

Reply:
It'd be awesome. We have to design the scan API nicely though.

Reply:
I am not sure about extending `TableScan` (can be convinced) but we can definitely create a utility class in core that will internally use `ManifestGroup`. I don' think we should expose this class. Maybe, it is safer to start with a utility and then see if it is something that can be part of `TableScan`.

Reply:
@aokolnychyi we are not talking about extending `TableScan`. We are talking about new scan interfaces to support incremental scan. Please see PR 4580 : https://github.com/apache/iceberg/pull/4580#discussion_r856499496